src/WinRPM.jl

```diff
@@ -46,33 +46,54 @@ function __init__()
     update(false, false)
 end
 
-if isunix()
-    function download(source::AbstractString)
-        x = HTTPC.get(source)
-        unsafe_string(x.body), x.http_code
+if VERSION < v"0.7.0-DEV"
+    if isunix()
+        function download(source::AbstractString)
+            x = HTTPC.get(source)
+            unsafe_string(x.body), x.http_code
+        end
+    elseif iswindows()
+        function download(source::AbstractString; retry=5)
+            dest = Vector{UInt16}(261)
+            for i in 1:retry
+                res = ccall((:URLDownloadToCacheFileW, :urlmon), stdcall, Cuint,
+                (Ptr{Void}, Ptr{UInt16}, Ptr{UInt16}, Clong, Cint, Ptr{Void}),
+                C_NULL, transcode(UInt16, source), dest, sizeof(dest) >> 1, 0, C_NULL)
+                if res == 0
+                    resize!(dest, findfirst(iszero, dest) - 1)
+                    filename = transcode(String, dest)
+                    if isfile(filename)
+                        return read(filename, String), 200
+                    end
+                else
+                    warn("Unknown download failure, error code: $res")
+                end
+                warn("Retry $i/$retry downloading: $source")                
+            end
+            return "", 0
+        end
+    else
+        error("Platform not supported: $(Sys.KERNEL)")
     end
-elseif iswindows()
-    function download(source::AbstractString; retry=5)
-        dest = Vector{UInt16}(261)
+else
+    function download(source::AbstractString; retry=5)        
         for i in 1:retry
-            res = ccall((:URLDownloadToCacheFileW, :urlmon), stdcall, Cuint,
-              (Ptr{Void}, Ptr{UInt16}, Ptr{UInt16}, Clong, Cint, Ptr{Void}),
-              C_NULL, transcode(UInt16, source), dest, sizeof(dest) >> 1, 0, C_NULL)
-            if res == 0
-                resize!(dest, findfirst(iszero, dest) - 1)
-                filename = transcode(String, dest)
+            try
+                filename = joinpath(tempdir(), split(source, "/")[end])
```

Comment:
this looks highly likely to have collisions

Reply:
So you'd rather not enforce the filename?

Reply:
No, don't drop most of source.